### PR TITLE
Convert SEO plugin to strapi v5

### PR DESCRIPTION
**Pull request by jhoward1994**
### What does it do?

- Converts the SEO plugin to strapi v5

### Related issue(s)/PR(s)

https://strapi-inc.atlassian.net/browse/DX-1428	

Uses additional exports added to the monorepo to be tested with https://github.com/strapi/strapi/pull/21209 if not merged

To test:
- Clone this branch
- Run this in the plugin codebase `yarn; yarn run watch:link` - This will publish the package to yalc locally. (make sure to install yalc globally https://www.npmjs.com/package/yalc)
- In a strapi v5 app run `yarn dlx yalc add --link @strapi/plugin-seo && yarn install`. This will install the plugin as a dependency from yalc

DX-1428

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,29 +1,62 @@
...
+      "default": "./dist/server/index.js"
+    }
   },
-  "repository": {
-    "type": "git",
-    "url": "https://github.com/strapi/strapi-plugin-seo.git"
+  "files": [
+    "dist",
+    "./strapi-server.js"
+  ],
```
Comment (innerdvations) [inline package.json:25]:
```suggestion
  "files": [
    "dist"
  ],
```

File: admin/src/translations/en.json
Comment (innerdvations) [inline admin/src/translations/en.json:1]: Why are we deleting the translations?

**Review by innerdvations — COMMENTED** [review]
Overall looks ok, but I will do a more thorough review and QA once we release the next RC with your fixes for the missing exports

File: admin/src/translations/en.json
Comment (innerdvations) [reply admin/src/translations/en.json:1]: Might be related to this, but I got an error on the getstarted project about missing translations that prevented the admin from loading

**Comment by jhoward1994** [comment]
I've found some bugs in the injected CM SEO component - working through them now. Will set this ready to review once that is done

**Comment by MDSilviu** [comment]
Any ETA on this PR? As v5 of strapi is officially released i wonder when we can get this plugin working again with the new version

**Comment by namnhfreelancer** [comment]
any updates ?

**Comment by Mcastres** [comment]
Hello @jhoward1994 👋 
Thanks for working on this!

I added the plugin in the LaunchPad application. I have two feedback:

- It seems that there is a CSS issue concerning the SEO block component
Capture d’écran 2024-10-04 à 15 33 00

- When clicking on "See details", I have this error:
Capture d’écran 2024-10-04 à 15 32 49

**Comment by jhoward1994** [comment]
- When clicking on "See details", I have this error:
Can you confirm you've installed the plugin like this

- Clone this branch
- Run this in the plugin codebase yarn; yarn run watch:link - This will publish the package to yalc locally. (make sure to install yalc globally [npmjs.com/package/yalc](https://www.npmjs.com/package/yalc))
- In a strapi v5 app run yarn dlx yalc add --link @strapi/plugin-seo && yarn install. This will install the plugin as a dependency from yalc

You can see the docs here
https://docs.strapi.io/dev-docs/plugins/development/create-a-plugin#linking-the-plugin-to-your-project

This is known issue with strapi when running in production with locally installed plugins - the yalc installation approach above should work

**Comment by jhoward1994** [comment]
image

@Mcastres 

[`ffb5302` (#76)](https://github.com/strapi/strapi-plugin-seo/pull/76/commits/ffb53022c99307cc274e9a262f3658c4272c6770) Will fix the styling to look like the image above

(ignore the preview button, that is added in the monorepo example and nothing to do with the SEO plugin)

**Comment by NEDDL** [comment]
@innerdvations @Mcastres @Bassel17 any ETA for this merge?

**Review by Mcastres — COMMENTED** [review]
Thanks for the updates @jhoward1994 👏 

I made the plugin work by simply placing the plugin inside my Strapi app. The process you described above doesn't work for me even after cleaning/installing/building again :/

I would only have two feedback:

- Is it possible to give the plugin summary that is injected in the CM some margin-top? I find it too close to the Save button
- The icons are not colored in the modal (see screenshots)
Capture d’écran 2024-10-07 à 16 03 42

Otherwise it looks good to me! 🙏 
Thanks for working on this!

**Comment by jhoward1994** [comment]
Will do on the padding.

The icon color is a limitation on the design system. I will be making that change through https://github.com/strapi/design-system/pull/1794 but think we can continue with this in the meantime. Wdyt?

**Review by Mcastres — APPROVED** [review]
LGTM

**Comment by jhoward1994** [comment]
Tested via https://www.npmjs.com/package/@strapi/plugin-seo/v/2.0.0-rc.1

I will publish v2 to npm from `main` after merging this PR

**Pull request merged**